applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -348,17 +352,23 @@ void DistanceModificationProcess::RecoverDeactivationPreviousState(){
         auto it_elem = mrModelPart.ElementsBegin() + i_elem;
         it_elem->Set(ACTIVE,true);
     }
-
-    // Free the negative DOFs that were fixed
-    #pragma omp parallel for
-    for (int i_node = 0; i_node < static_cast<int>(mrModelPart.NumberOfNodes()); ++i_node){
-        auto it_node = mrModelPart.NodesBegin() + i_node;
-        if (it_node->GetValue(EMBEDDED_IS_ACTIVE) == 0){
-            // Free the nodal DOFs that were fixed
-            it_node->Free(PRESSURE);
-            it_node->Free(VELOCITY_X);
-            it_node->Free(VELOCITY_Y);
-            it_node->Free(VELOCITY_Z);
+    if ((mDoubleVariablesList.size() > 0.0) || (mComponentVariablesList.size() > 0.0)){
+        // Free the negative DOFs that were fixed
+        #pragma omp parallel for
+        for (int i_node = 0; i_node < static_cast<int>(mrModelPart.NumberOfNodes()); ++i_node){
+            auto it_node = mrModelPart.NodesBegin() + i_node;
+            if (it_node->GetValue(EMBEDDED_IS_ACTIVE) == 0){
+                for (int i_var = 0; i_var < static_cast<int>(mDoubleVariablesList.size()); i_var++){
+                    Variable<double> double_var = KratosComponents< Variable<double> >::Get(mDoubleVariablesList[i_var]);
```

Comment:
Do not access KratosComponents inside a node loop if you can avoid it, it is a relatively expensive call. I propose that you store list of variables instead of list of strings in a separate comment but, even if you don't do that, call KratosComponents once, outside the loop.